elements/drawtools/src/methods/draw/emit-drawn-features.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Emits drawn features after a timeout to allow updating drawn features.
+ *
+ * @param {import("../../main").EOxDrawTools} EoxDrawTool - The drawing tool instance.
+ */
+const emitDrawnFeaturesMethod = (EoxDrawTool) => {
+  // Function to emit features after a timeout (ensures update)
+  const emitFeatures = () => {
+    // Update drawnFeatures with features from drawLayer's source
+    EoxDrawTool.drawnFeatures = EoxDrawTool.drawLayer.getSource().getFeatures();
+    EoxDrawTool.requestUpdate();
+
+    // Dispatch a 'drawupdate' event with drawn features as detail
+    EoxDrawTool.dispatchEvent(
```

Comment:
There seems to be some issue with our custom element parser, as it doesn't recognize the emitted event in this case. If you compare with e.g. `drawtools`, it does if in the main.js file it directly does something like
```
  #emitDrawnFeatures() {
    ...
    this.dispatchEvent(
      new CustomEvent("drawupdate", {
        detail: EoxDrawTool.drawnFeatures,
      })
    );
  }
```
This could be mitigated by moving the dispatch to main.js, or maybe there is some other solution. The ideal outcome would show the event in the docs:

![image](https://github.com/EOX-A/EOxElements/assets/26576876/c5f818d8-1cec-4de9-874c-b38fc17ab8c2)

Reply:
Thanks for noticing this out. I will try to find an optimal work around for this.

Reply:
Done ✅